concurrent_map.go

```diff
@@ -290,7 +316,27 @@ func (m ConcurrentMap) Keys() []string {
 	return keys
 }
 
-//Reviles ConcurrentMap "private" variables to json marshal.
+// FindOldest search for the oldest key in the key slice.
+func (m ConcurrentMap) FindOldest() string {
+	keys := []string{}
+	for i := 0; i < ShardCount; i++ {
```

Comment:
We are not necessarily interested in finding the oldest in the entire map if this causes lot of processing, so no need to iterate over all shards.
I would rather find the oldest inside the shard I am adding to, if this shard is full, and remove that.